apollo-router/src/plugins/connectors/tests/mod.rs

```diff
@@ -731,23 +731,159 @@ async fn test_override_headers_with_config() {
                 .method("GET")
                 .header(
                     HeaderName::from_str("x-forward").unwrap(),
-                    HeaderValue::from_str("forwarded").unwrap(),
+                    HeaderValue::from_str("forwarded-by-config").unwrap(),
+                )
+                .header(
+                    HeaderName::from_str("x-insert").unwrap(),
+                    HeaderValue::from_str("inserted-by-config").unwrap(),
+                )
+                .header(
+                    HeaderName::from_str("x-insert-multi-value").unwrap(),
+                    HeaderValue::from_str("third,fourth").unwrap(),
                 )
+                .path("/users"),
+        ],
+    );
+}
+
+#[tokio::test]
+async fn should_only_send_named_header_once_when_both_config_and_schema_propagate_header() {
+    let mock_server = MockServer::start().await;
+    mock_api::users().mount(&mock_server).await;
+
+    execute(
+        STEEL_THREAD_SCHEMA,
+        &mock_server.uri(),
+        "query { users { id } }",
+        Default::default(),
+        Some(json!({
+            "connectors": {
+                "subgraphs": {
+                    "connectors": {
+                        "$config": {
+                          "source": {
+                            "val": "val-from-config-source"
+                          },
+                          "connect": {
+                            "val": "val-from-config-connect"
+                          },
+                        }
+                    }
+                }
+            },
+            "headers": {
+              "connector": {
+                "all": {
+                  "request": [
+                  {
+                    "propagate": {
+                      "named": "x-forward",
```

Comment:
should we also have a insert rule test, and a propagate matching test? mostly as a regression check

Reply:
what happens if there's `headers: [{ name: "x-foo", value: "bar" }]` in SDL, and `remove: { named: "x-foo" }` in yaml?